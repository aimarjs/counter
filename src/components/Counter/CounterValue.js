import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const CounterValue = ({ value }) => (
  <Value>{value}</Value>
);

const Value = styled.h2`
  font-size: 64px;
`;

CounterValue.propTypes = {
  value: PropTypes.number.isRequired,
};

export default CounterValue;
