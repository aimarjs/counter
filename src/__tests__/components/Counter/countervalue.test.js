import React from 'react';
import renderer from 'react-test-renderer';

import CounterValue from '../../../components/counter/CounterValue';

describe('CounterValue Component', () => {
  it('capturing snapshot of CounterValue', () => {
    const value = 0;

    const renderedValue = renderer.create(
      <CounterValue
        value={value}
      />,
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
