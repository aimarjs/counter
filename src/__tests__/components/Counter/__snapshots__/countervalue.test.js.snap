// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterValue Component capturing snapshot of CounterValue 1`] = `
<h2
  className="sc-bdVaJa jKoFUU"
>
  0
</h2>
`;
